File: tests/fixtures/edgecase_repo/javascript/src/module6.js
const { Service5 } = require("./module5");

class Base6 {
  constructor(name) {
    this.name = name;
  }
}

class Service6 extends Base6 {
  handle(payload) {
    this.validate(payload);
    return new Service5("x").handle(payload);
  }

  validate(payload) {
    return payload !== null;
  }
}

function helper6(value) {
  return value;
}

module.exports = { Service6 };

File: tests/fixtures/edgecase_repo/rust/src/module6.rs
use crate::module5::Service5;

pub struct Base6 {
    pub name: String,
}

pub struct Service6 {
    pub base: Base6,
}

pub trait Handler6 {
    fn handle(&self, payload: &str) -> bool;
}

impl Handler6 for Service6 {
    fn handle(&self, payload: &str) -> bool {
        self.validate(payload);
        true
    }
}

impl Service6 {
    fn validate(&self, payload: &str) {}
}

pub fn helper_6(value: &str) -> &str {
    value
}
